src/dsp_tools/commands/xmlupload/iiif_client.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass
+
+import requests
+
+from dsp_tools.commands.xmlupload.models.input_problems import IIIFUriProblem
+
+
+@dataclass(frozen=True)
+class IIIFUriValidator:
+    """Client handling communication with external IIIF-servers to do a health check."""
+
+    uri: str
+    regex_has_passed: bool
```

Comment:
The reason why it is on the level of a single URI rather than all the URIs is because as opposed to the Connection class, we probably speak to different APIs meaning that there is no commonality of the connection... But if you feel strongly about this I can change it.